zjit/src/hir.rs

```diff
@@ -1020,6 +1020,30 @@ impl Insn {
 pub struct InsnPrinter<'a> {
     inner: Insn,
     ptr_map: &'a PtrPrintMap,
+    iseq: Option<IseqPtr>,
+}
+
+/// Get the name of a local variable given iseq, level, and ep_offset.
+/// Returns
+/// - `:name, ` if iseq is available and name is a real identifier,
+/// - `<anon>, ` for anonymous locals.
+/// - empty string if iseq is not available.
```

Comment:
Could you let the comment also explain when this happens? I'm not sure why we need to support this. Is it used only for Rust-level unit tests?

Reply:
Sure. It's used in HIR dump, graphviz output, and iongraph output.

Reply:
Oh I thought you meant the return value of this function in general. 

`Insn` can also be printed in panic/debug messages too, like in `disasm.rs` we have:

```rs
    // For each instruction in this block
    for insn in insns.as_ref() {
        // ...
        writeln!(&mut out, "  {}", format!("{insn}").trim()).unwrap();
    }
```

In these cases, I think the `fmt` implementation for `Display` trait would be called, which can't access iseq:

```rs
impl std::fmt::Display for Insn {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        self.print(&PtrPrintMap::identity(), None).fmt(f)
    }
}
```

So the empty string is for those cases.